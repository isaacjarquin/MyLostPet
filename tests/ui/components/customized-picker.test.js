import "react-native"
import React from "react"
import SecondaryMenuModal from "../../../components/customized-picker"
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import renderer from "react-test-renderer"

describe("SecondaryMenuModal", () => {
    const navigate = jest.fn()

    const props = {
        handler: jest.fn(),
        hidePetTypeModal: jest.fn(),
        items: []
    };

    beforeEach(() => {
        wrapper = shallow(<SecondaryMenuModal {...props} />);
    });

    it("renders correctly", () => {
        expect(wrapper).toBeDefined()
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    describe("handleSelected", () => {
        beforeEach(() => {
            wrapper.instance().handleSelected()
        })

        it("navigate to MissingPetForm", () => {
            expect(props.handler).toHaveBeenCalledTimes(1)
            expect(props.hidePetTypeModal).toHaveBeenCalledTimes(1)
            expect(props.handler).toHaveBeenCalledWith("")
        })
    })
})