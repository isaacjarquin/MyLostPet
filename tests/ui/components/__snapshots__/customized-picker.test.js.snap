// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryMenuModal renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "marginTop": "75%",
      }
    }
    useNativeDriver={false}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        color="grey"
        name="keyboard-arrow-down"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={30}
        type="MaterialIcons"
        underlayColor="white"
      />
    </TouchableOpacity>
    <Picker
      mode="dialog"
      onValueChange={[Function]}
      selectedValue=""
      style={
        Object {
          "borderColor": "grey",
          "borderRadius": 10,
          "borderWidth": 1,
          "margin": 10,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
          "borderRadius": 10,
          "margin": 10,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
          }
        }
      >
        Confirmar
      </Text>
    </TouchableOpacity>
  </ReactNativeModal>
</View>
`;
